Mathlib/NumberTheory/Transcendental/AnalyticPart.lean

```diff
@@ -0,0 +1,606 @@
+/-
+Copyright (c) 2025 Michail Karatarakis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michail Karatarakis
+-/
+module
+
+public import Mathlib.Analysis.Analytic.Order
+public import Mathlib.Analysis.Complex.CauchyIntegral
+public import Mathlib.Analysis.Normed.Module.Connected
+
+/-!
+Some auxiliary lemmata covering the analytic part of the proof of the Gelfond–Schneider theorem.
+-/
+
+set_option autoImplicit true
+set_option linter.style.multiGoal false
+set_option linter.unusedVariables false
+
+@[expose] public section
+
+open BigOperators Module.Free Fintype Matrix Set Polynomial Finset Complex Differentiable AnalyticAt
+
+lemma exists_mem_finset_min' {γ : Type _} {β : Type _} [LinearOrder γ]
+  (s : Finset β) (f : β → γ) (Hs : s.Nonempty) :
+  ∃ x ∈ s, ∃ y, y = f x ∧ ∀ x' ∈ s, y ≤ f x' := by
+  obtain ⟨x, hxs, hx⟩ := s.exists_min_image f Hs
+  exact ⟨x, hxs, f x, rfl, hx⟩
+
+lemma AnalyticOnEq (f g : ℂ → ℂ) (U : Set ℂ) :
+  (∀ z ∈ U, f z = g z) → AnalyticOn ℂ f U → AnalyticOn ℂ g U := by
+    intros Heq HA
+    intro x Hx
+    obtain ⟨p, renn, H⟩ := HA x Hx
+    use p, renn
+    constructor
+    · exact H.r_le
+    · exact H.r_pos
+    · have : ∀ {y : ℂ}, x + y ∈ insert x U →
+      y ∈ EMetric.ball 0 renn → HasSum (fun n ↦ (p n) fun x ↦ y) (f (x + y)) := H.hasSum
+      unfold HasSum at this ⊢
+      have Hinsert : insert x U = U := by simp_all [Set.insert_eq_of_mem]
+      rw [Hinsert] at this ⊢
+      intros y Hxy
+      have := this Hxy
+      rwa [← Heq _ Hxy]
+
+
+lemma AnalyticOnEquiv (f g : ℂ → ℂ) (U : Set ℂ) :
```

Comment:
This should be called `analyticOn_congr` and should go right below `AnalyticOn.congr`.